executor/cook/io_helper.py

```diff
@@ -149,7 +149,7 @@ def trigger_flush_daemon():
         logging.info('Starting to pipe {}'.format(label))
         trigger_flush_daemon()
         while True:
-            line = out_file.readline(max_bytes_read_per_line)
+            line = out_buffer.read1(max_bytes_per_read)
```

Comment:
:+1: Based on the documentation you linked to, using `read1` definitely sounds like an improvement to me.